### PR TITLE
Get book data from dir refactor

### PR DESCRIPTION
**Pull request by cassieesposito**
Minor tweaks to the way scanner handles information about series volumes.

### CODE REVIEW

File: server/utils/scandir.js [hunk 1]
```diff
@@ -209,91 +209,23 @@ function getBookDataFromDir(folderPath, relPath, parseSubtitle = false) {
...
-  // Subtitle is everything after " - "
-  var subtitle = null
-  if (parseSubtitle && title.includes(' - ')) {
-    var splitOnSubtitle = title.split(' - ')
-    title = splitOnSubtitle.shift()
-    subtitle = splitOnSubtitle.join(' - ')
-  }
+  // The  may contain various other pieces of metadata, these functions extract it.
+  var [folder, narrators] = getNarrator(folder)
+  if (series) { var [folder, sequence] = getSequence(folder) }
```
Comment (advplyr) [inline server/utils/scandir.js:218]: Does this work? Folder and sequence are being defined inside the if statement scope.

Comment (advplyr) [reply server/utils/scandir.js:218]: I don't think I ever new that variables defined in if statements are available outside, but it appears to be so. "Function scope" they say.

Comment (cassieesposito) [reply server/utils/scandir.js:218]: I believe this is one of the differences between var and let.

Comment (cassieesposito) [reply server/utils/scandir.js:218]: Yep.

Welcome to Node.js v16.15.0.
Type ".help" for more information.
> if (1) {var w = 'hello'}
undefined
> if (1) {let x = 'world'}
undefined
> w
'hello'
> x
Uncaught ReferenceError: x is not defined
>

Comment (advplyr) [reply server/utils/scandir.js:218]: Probably best to avoid putting this in that scope in the future, might cause some confusion for others too.

**Pull request merged**